controllers/secret_controller.go

```diff
@@ -217,42 +151,82 @@ func (r *ReconcileSecret) Reconcile(request reconcile.Request) (reconcile.Result
 	}
 }
 
-// removeInvalidAnnotationsFromLinuxNodes corrects annotations applied by previous versions of WMCO.
-func (r *ReconcileSecret) removeInvalidAnnotationsFromLinuxNodes() error {
-	nodes := &core.NodeList{}
-	err := r.client.List(context.TODO(), nodes, client.MatchingLabels{core.LabelOSStable: "linux"})
+// RemoveInvalidAnnotationsFromLinuxNodes makes a best effort to correct annotations applied by previous versions of WMCO.
+func (r *SecretReconciler) RemoveInvalidAnnotationsFromLinuxNodes(config *rest.Config) {
+	// create a new clientset as this function will be called before the manager's client is started
+	kc, err := kubernetes.NewForConfig(config)
 	if err != nil {
-		return errors.Wrapf(err, "error getting node list")
+		r.log.Error(err, "error creating kubernetes clientset")
+	}
+
+	nodes, err := kc.CoreV1().Nodes().List(context.TODO(), meta.ListOptions{LabelSelector: core.LabelOSStable + "=linux"})
+	if err != nil {
+		r.log.Error(err, "error getting node list")
 	}
 	// The public key hash was accidentally added to Linux nodes in WMCO 2.0 and must be removed.
 	// The `/` in the annotation key needs to be escaped in order to not be considered a "directory" in the path.
 	escapedPubKeyAnnotation := strings.Replace(nodeconfig.PubKeyHashAnnotation, "/", "~1", -1)
 	patchData := fmt.Sprintf(`[{"op":"remove","path":"/metadata/annotations/%s"}]`, escapedPubKeyAnnotation)
 	for _, node := range nodes.Items {
 		if _, present := node.Annotations[nodeconfig.PubKeyHashAnnotation]; present == true {
-			err = r.client.Patch(context.TODO(), &node, client.RawPatch(kubeTypes.JSONPatchType, []byte(patchData)))
+			_, err = kc.CoreV1().Nodes().Patch(context.TODO(), node.GetName(), kubeTypes.JSONPatchType,
+				[]byte(patchData), meta.PatchOptions{})
 			if err != nil {
-				return errors.Wrapf(err, "error removing public key annotation from node %s", node.GetName())
+				r.log.Error(err, "error removing public key annotation from node", "node", node.GetName())
```

Comment:
can create an event for this.

Reply:
I mean leave the function as is without trying to fix the fact that the returned error is not handled at the call site.